xchange-ascendex/src/main/java/org/knowm/xchange/ascendex/AscendexAdapters.java

```diff
@@ -0,0 +1,255 @@
+      List<AscendexCashAccountBalanceDto> ascendexCashAccountBalanceDtoList) {
+    List<Balance> balances = new ArrayList<>();
+
+    ascendexCashAccountBalanceDtoList.forEach(
+            ascendexCashAccountBalanceDto ->
+            balances.add(
+                new Balance.Builder()
+                    .currency(new Currency(ascendexCashAccountBalanceDto.getAsset()))
+                    .available(ascendexCashAccountBalanceDto.getAvailableBalance())
+                    .total(ascendexCashAccountBalanceDto.getTotalBalance())
+                    .frozen(
+                        ascendexCashAccountBalanceDto
+                            .getTotalBalance()
+                            .subtract(ascendexCashAccountBalanceDto.getAvailableBalance()))
+                    .build()));
+
+    return new AccountInfo(
+        Wallet.Builder.from(balances)
+            .id("spot")
+            .features(new HashSet<>(Collections.singletonList(Wallet.WalletFeature.TRADING)))
+            .build());
+  }
+
+  public static AscendexPlaceOrderRequestPayload adaptLimitOrderToAscendexPlaceOrderRequestPayload(
+      LimitOrder limitOrder) {
+    return new AscendexPlaceOrderRequestPayload(
+        limitOrder.getInstrument().toString(),
+        Date.from(Instant.now()).toInstant().toEpochMilli(),
+        limitOrder.getOriginalAmount().toString(),
+        AscendexPlaceOrderRequestPayload.AscendexOrderType.limit,
+        limitOrder.getType().equals(Order.OrderType.ASK)
+            ? AscendexPlaceOrderRequestPayload.AscendexSide.sell
+            : AscendexPlaceOrderRequestPayload.AscendexSide.buy,
+        null,
+        limitOrder.getLimitPrice().toString(),
+        null,
+        limitOrder.hasFlag(AscendexFlags.POST_ONLY),
+        null,
+        null);
+  }
+
+  public static UserTrades adaptUserTrades(
+      List<AscendexOpenOrdersResponse> ascendexOrderHistoryResponse) {
+    List<UserTrade> userTrades = new ArrayList<>();
+
+    ascendexOrderHistoryResponse.forEach(
+        order ->
+            userTrades.add(
+                new UserTrade.Builder()
+                    .feeAmount(order.getCumFee())
+                    .orderId(order.getOrderId())
+                    .price(order.getPrice())
+                    .type(adaptAscendexSideToOrderType(order.getSide()))
+                    .originalAmount(order.getOrderQty())
+                    .id(order.getOrderId())
+                    .timestamp(order.getLastExecTime())
+                    .currencyPair(
+                        CurrencyPairDeserializer.getCurrencyPairFromString(order.getSymbol()))
+                    .feeCurrency(new Currency(order.getFeeAsset()))
+                    .instrument(
+                        CurrencyPairDeserializer.getCurrencyPairFromString(order.getSymbol()))
+                    .build()));
+
+    return new UserTrades(userTrades, Trades.TradeSortType.SortByTimestamp);
+  }
+
+  public static OpenOrders adaptOpenOrders(
```

Comment:
can this + adaptOpenOrderById be condensed into a single method ?

Reply:
Done